openmdao/recorders/sqlite_reader.py

```diff
@@ -358,9 +381,16 @@ def list_source_vars(self, source):
         if case.residuals:
             dct['residuals'] = list(case.residuals)
 
-        return dct
+        if not out_stream:
+            return dct
+        else:
+            if out_stream is _DEFAULT_OUT_STREAM:
+                out_stream = sys.stdout
 
-    def list_cases(self, source=None, recurse=True, flat=True):
+            if out_stream:
+                return write_source_table(dct, out_stream)
```

Comment:
I think this should always have a return value (the `dct`) regardless of the out_stream, similar to the `list_` functions on System (and in keeping with the docstring)
The `write_source_table` function does not return anything.